File: project/soyoung-onboarding/src/main/java/com/survey/soyoung_onboarding/entity/Survey.java
package com.survey.soyoung_onboarding.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
public class Survey {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    
    /**
     * 설문 제목
     */
    private String title;

    /**
     * 설문 설명
     */
    private String description;

    /**
     *  질문 항목
     */
    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Question> questions = new ArrayList<>();

    private Date reg_date;

    private Date update_date;

}

File: project/soojinhan-onboarding/src/main/kotlin/onboardingproject/project/domain/Survey.kt
package onboardingproject.project.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import onboardingproject.project.common.domain.BaseEntity
import java.util.*

/**
 * packageName : onboardingproject.project.domain
 * fileName    : Survey
 * author      : hsj
 * date        : 2025. 6. 14.
 * description : 설문조사
 */
@Entity
class Survey(
    @Id
    val id: String = UUID.randomUUID().toString(),

    @Column(nullable = false)
    var title: String,

    @Column
    var description: String?,
) : BaseEntity()